src/objective-c/manual_tests/GrpcIosTestUITests/GrpcIosTestUITests.m

```diff
@@ -42,21 +47,44 @@ - (void)setUp {
   XCTAssert([settingsApp.navigationBars[@"Settings"] waitForExistenceWithTimeout:kWaitTime]);
   // Turn off airplane mode
   [self setAirplaneMode:NO];
+
+  // Turn on wifi
+  [self setWifi:YES];
 }
 
 - (void)tearDown {
 }
 
 - (void)doUnaryCall {
   [testApp activate];
-  [testApp.buttons[@"Unary call"] tap];
+  [self pressButton:@"Unary call"];
+}
+
+- (void)do10UnaryCalls {
+  [testApp activate];
+  [self pressButton:@"10 Unary calls"];
+}
+
+
+- (void)pressButton: (NSString *)name {
+  // Wait for button to be visible
+  while (![testApp.buttons[name] exists] ||
+         ![testApp.buttons[name] isHittable]) {
+    [NSThread sleepForTimeInterval:1];
+
+  }
+  // Wait until all events in run loop have been processed
+  while (CFRunLoopRunInMode(kCFRunLoopDefaultMode, 0.1, true) == kCFRunLoopRunHandledSource);
+
+  NSLog(@"Pressing button: %@", name);
+  [testApp.buttons[name] tap];
 }
 
 - (void)doStreamingCall {
   [testApp activate];
-  [testApp.buttons[@"Start streaming call"] tap];
-  [testApp.buttons[@"Send Message"] tap];
-  [testApp.buttons[@"Stop streaming call"] tap];
+  [self pressButton:@"Start streaming call"];
+  [self pressButton:@"Send Message"];
+  [self pressButton:@"Stop streaming call"];
```

Comment:
Fair point :). I guess I can remove some of these test cases where unary and streaming call are effectively the same thing since there is no network flap while the streaming call is in progress.